src/interfaces/chain.cpp

```diff
@@ -234,6 +225,15 @@ class ChainImpl : public Chain
         std::unique_ptr<Chain::Lock> result = std::move(lock); // Temporary to avoid CWG 1579
         return result;
     }
+    Optional<int> getHeight() override
+    {
+        WAIT_LOCK(cs_main, lock);
```

Comment:
In commit "[wallet] Move getHeight from Chain::Lock interface to simple Chain" (b4eb6aee0d259c3586894694a67928759503c823)

This is fine, but in this commit and other commits throughout this PR could simplify `WAIT_LOCK(cs_main, lock)` to just `LOCK(cs_main)` since `lock` object is unused

Reply:
Right, only difference between them is what is logged for debug, attributing a name for the lock for `WAIT_LOCK(cs_main, lock)`? They both declare `DebugLock<decltype>`. Updating